torchx/lib/torchx/backend.ex

```diff
@@ -921,6 +921,29 @@ defmodule Torchx.Backend do
     |> to_nx(out)
   end
 
+  def dot(
+        %T{type: out_type} = out,
+        %T{type: left_type, data: %TB{ref: left_ref}},
+        left_axes,
+        left_batched_axes,
+        %T{type: right_type, data: %TB{ref: right_ref}},
+        right_axes,
+        right_batched_axes
+      ) do
+    left_axes = Enum.map(left_axes, fn x -> x - Enum.count(left_batched_axes, &(&1 < x)) end)
+    right_axes = Enum.map(right_axes, fn x -> x - Enum.count(right_batched_axes, &(&1 < x)) end)
+
+    Torchx.batched_tensordot(
```

Comment:
Should I try to make tensordot optionally aware of batching instead of exposing a separate function?

Reply:
I think it is fine as is, I would just remove the `batched_` prefix.

Reply:
I ended up merging all into a single tensordot/6 clause for the backend and a single function at the cpp level